feat: add erroring out / refactor / add tests

cdrage opened the pull request:
feat: add erroring out / refactor / add tests

### What does this PR do?

* Checks to see if the image has actually errored out before continuing
* Moves the "container utility" functions to a separate file
* Adds a lot more tests

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

N/A

### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/38

### How to test this PR?

1. Start a build image
2. Stop the container during the build process.
3. It will fail as the container image will exit with error (1), rather
   than saying success

<!-- Please explain steps to reproduce -->

cdrage commented:
In draft FYI! Will ping when in review.

@deboer-tim Letting you know in case this gets merged in and you have to rebase lots, this moves stuff to a different file + adds more tests.

cdrage commented:
Ready for review!

cdrage commented on src/build-disk-image.ts:
```diff
@@ -104,61 +103,68 @@ export async function buildDiskImage(image: any) {
       telemetryLogger.logUsage('buildDiskImage', telemetryData);
 
-      // Only if success = true and type = ami
-      if ((successful && selectedType === 'ami') || selectedType === 'raw') {
```
See if you can spot the logic error / hack for the demo haha. This part I fixed up since it kept saying success if you did raw...

cdrage commented on src/container-utils.ts:
```diff
@@ -0,0 +1,117 @@
+        if (container.Id === containerId && container.State === 'exited') {
+          // Let's stop the loop if the container has exited / stopped
+          containerRunning = false;
+          // Container.status reports "ex. Exited (1) Less than a second ago" when it
```
Will now check if Exited (0) happens.

cdrage replied:
You can trigger this by simply "stopping" the container mid-build.

cdrage commented:
~EDIT: Found a few more bugs~

Had an issue with removing the previous container build, since I was passing in the container name rather than ID, that's fixed now.

It's ready for review!

deboer-tim commented on src/build-disk-image.ts:
```diff
@@ -21,10 +21,10 @@ import * as extensionApi from '@podman-desktop/api';
 const bootcImageBuilderContainerName = '-bootc-image-builder';
-const bootcImageBuilderName = 'quay.io/centos-bootc/bootc-image-builder:latest-1704948606';
-let diskImageBuildingName: string;
+const bootcImageBuilderName = 'quay.io/centos-bootc/bootc-image-builder:latest';
```
I don't think I'm comfortable with this yet - changing one argument could easily break our extension and they're likely not trying to maintain compatibility yet. Can we stick with a tagged version or sha, at least for now?

benoitf replied:
you might pin the release to a given sha but add an option that will be in preferences where as a user you might/could override or a checkbox 'dev/stable' or drop-down 'latest release/ dev release or extension's versions

So default is locked but you allow to test other versions

cdrage commented:
@deboer-tim I've moved the constant variables to `constants.ts` so it's in a more unified place since we may be changing that section a lot as we update bootc-image-builder.

I've updated the image to the latest tag that has the `.iso` support / latest changes.
